File: FunctionSampleDir/ListItemDragProject/app/src/main/java/com/mysample/javadragapplication/MainActivity.java
package com.mysample.javadragapplication;

import android.content.Context;
import android.graphics.Bitmap;
import android.net.Uri;
import android.provider.MediaStore;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.helper.ItemTouchHelper;
import android.view.View;
import android.widget.Button;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.target.BitmapImageViewTarget;

import java.io.ByteArrayOutputStream;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity
{
    RecyclerView recyclerView;
    RecyclerViewAdapter mAdapter;
    ArrayList<String> stringArrayList = new ArrayList<>();
    ItemTouchHelper touchHelper;

    Button button;

    boolean isEnableAttach = false;

    ItemTouchHelper.Callback callback;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerView);
        button = findViewById(R.id.button);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                isEnableAttach = !isEnableAttach;
                if(isEnableAttach)
                {
                    button.setText("저장");
                    ((ItemMoveCallback) callback).setDraggable(false);
                } else {
                    button.setText("순서변경하기");
                    touchHelper.attachToRecyclerView(recyclerView);
                    ((ItemMoveCallback) callback).setDraggable(true);
                }
            }
        });


        populateRecyclerView();
    }

    private void populateRecyclerView() {
        stringArrayList.add("Item 1");
        stringArrayList.add("Item 2");
        stringArrayList.add("Item 3");
        stringArrayList.add("Item 4");
        stringArrayList.add("Item 5");
        stringArrayList.add("Item 6");
        stringArrayList.add("Item 7");
        stringArrayList.add("Item 8");
        stringArrayList.add("Item 9");
        stringArrayList.add("Item 10");

        mAdapter = new RecyclerViewAdapter(stringArrayList,startDragListener);
        callback = new ItemMoveCallback(mAdapter);
        ((ItemMoveCallback) callback).setDraggable(true);
        touchHelper = new ItemTouchHelper(callback);
        touchHelper.attachToRecyclerView(recyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setAdapter(mAdapter);

        new Thread(new Runnable() {
            @Override
            public void run() {

            }
        }).start();
    }

    StartDragListener startDragListener = new StartDragListener() {
        @Override
        public void requestDrag(RecyclerView.ViewHolder viewHolder) {
            touchHelper.startDrag(viewHolder);
        }
    };

    public Uri getImageUri(Context inContext, Bitmap inImage)
    {
        ByteArrayOutputStream bytes = new ByteArrayOutputStream();
        inImage.compress(Bitmap.CompressFormat.JPEG, 100, bytes);
        String path = MediaStore.Images.Media.insertImage(inContext.getContentResolver(), inImage, "Title", null);
        return Uri.parse(path);
    }
}

File: FunctionSampleDir/TsDebugSample/app/src/main/java/com/example/tsdebugsample/MainActivity.kt
package com.example.tsdebugsample

import android.media.MediaPlayer
import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity()
{
    val TAG = "main"

    private var mp1 : MediaPlayer? = null
    private var mp2 : MediaPlayer? = null
    private var mp3 : MediaPlayer? = null

    override fun onCreate(savedInstanceState: Bundle?)
    {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mp1 = MediaPlayer.create(this, R.raw.a) // ts 파일 채널 1
        mp1?.setOnErrorListener { mediaPlayer, i, i2 ->
            Log.d(TAG, "onCreate: ")
            return@setOnErrorListener false
        }

        mp2 = MediaPlayer.create(this, R.raw.b) // ts 파일 채널 2
        mp2?.setOnErrorListener { mediaPlayer, i, i2 ->
            Log.d(TAG, "onCreate: ")
            return@setOnErrorListener false
        }

        mp3 = MediaPlayer.create(this, R.raw.c) // a 채널 2로 변경한 것
        mp3?.setOnErrorListener { mediaPlayer, i, i2 ->
            Log.d(TAG, "onCreate: ")
            return@setOnErrorListener false
        }
    }

    private fun test1(view: View)
    {
        mp1?.start()
    }
    private fun test2(view: View)
    {
        mp2?.start()
    }
    private fun test3(view: View)
    {
        mp3?.start()
    }
}